site/app/controllers/forum/ForumController.php

```diff
@@ -1363,57 +1364,62 @@ public function getSplitPostInfo() {
+            $tmp['is_staff_post'] = $version->getEditAuthor()->accessFullGrading();
+            $tmp['post_time'] = DateUtils::parseDateTime($version->getEditTimestamp(), $this->core->getConfig()->getTimezone())->format("n/j g:i A");
+            $output[] = $tmp;
             $edit_id++;
         }
         if (count($output) == 0) {
-            // Current post
-            $_post['user'] = !$this->modifyAnonymous($oc) && $anon ? '' : $oc;
-            $_post['content'] = $this->core->getOutput()->renderTwigTemplate("forum/RenderPost.twig", [
-                "post_content" => $current_post["content"],
+            // No history, get current post
+            $tmp = [];
+            $tmp['user'] = (!$this->modifyAnonymous($post->getAuthor()->getId()) && $post->isAnonymous()) ? '' : $post->getAuthor()->getId();
+            $tmp['content'] = $this->core->getOutput()->renderTwigTemplate("forum/RenderPost.twig", [
+                "post_content" => $post->getContent(),
                 "render_markdown" => false,
                 "post_attachment" => ForumUtils::getForumAttachments(
                     $post_id,
-                    $current_post['thread_id'],
-                    array_values($post_attachments[$post_id])[0],
+                    $post->getThread()->getId(),
+                    $post->getAttachments()->filter(function ($x) {
+                        return $x->isCurrent();
+                    })->map(function ($x) {
+                        return $x->getFileName();
+                    })->toArray(),
                     $this->core->getConfig()->getCoursePath(),
                     $this->core->buildCourseUrl(['display_file'])
                 ),
                 "edit_id" => $post_id . "-" . $edit_id,
             ]);
-            $_post['post_time'] = DateUtils::parseDateTime($current_post['timestamp'], $this->core->getConfig()->getTimezone())->format("n/j g:i A");
-            $output[] = $_post;
-        }
-        // Fetch additional information
-        foreach ($output as &$_post) {
-            $emptyUser = empty($_post['user']);
-            $_post['user_info'] = $emptyUser ? ['given_name' => 'Anonymous', 'family_name' => '', 'email' => '', 'pronouns' => '', 'display_pronouns' => false ] : $this->core->getQueries()->getDisplayUserInfoFromUserId($_post['user']);
-            $_post['is_staff_post'] = $emptyUser ? false : $this->core->getQueries()->isStaffPost($_post['user']);
+            $emptyAuthor = $tmp['user'] === '';
+            $tmp['user_info'] = $emptyAuthor ? ['given_name' => 'Anonymous', 'family_name' => '', 'email' => '', 'pronouns' => '', 'display_pronouns' => false ] : $post->getAuthor()->getDisplayInfo();
+            $tmp['is_staff_post'] = !$emptyAuthor && $post->getAuthor()->accessFullGrading();
+            $tmp['post_time'] = DateUtils::parseDateTime($post->getTimestamp(), $this->core->getConfig()->getTimezone())->format("n/j g:i A");
```

Comment:
The format string should be stored in a constant somewhere instead of repeating it multiple times throughout this file.